packages/cli/src/oclif/commands/jobs.js

```diff
@@ -0,0 +1,82 @@
+const BaseCommand = require('../ZapierBaseCommand');
+
+const { listMigrations } = require('../../utils/api');
+const { chain } = require('lodash');
+
+const getVersion = (versionStr) => versionStr.split('@')[1];
+
+class JobsCommand extends BaseCommand {
+  async perform() {
+    /**
+     * Migrations and Jobs are used somewhat interchangeably here.
+     * Migrations represents the collection of promotion and migration
+     * background "jobs" that were recently executed on this app.
+     */
+
+    this.startSpinner('Loading jobs');
+    const { migrations } = await listMigrations();
+    this.stopSpinner();
+
+    const jobs = chain(migrations)
+      .filter(
+        (migration) =>
+          migration.job_kind === 'migrate' || migration.job_kind === 'promote'
+      )
+      .map((migration) => {
+        const stepTotals = {
+          finished: 0,
+          estimating: 0,
+          skipped: 0,
+        };
+        for (const step of migration.progress.steps) {
+          if (step.status === 'success') {
+            stepTotals.finished++;
+          } else if (step.status === 'in_progress') {
+            stepTotals.estimating++;
+          } else if (step.status === 'skipped') {
```

Comment:
This might need more steps or a fallback. The big one that jumps out is that we're not accounting for `pending` steps. You might sync up with Season or Chang-Hung to learn more about how we count and track steps (and how we did this work in godzilla)

Reply:
Good point, just peeking at the App Migration System Monitor, counting/tracking steps may only pertain to the current step?
See [monitor](https://github.com/zapier/zapier/blob/develop/web/backend/zapier/migration_fsm/templates/migration_fsm/monitor.html#L125) and [dev-platform](https://gitlab.com/zapier/team-developer-platform/dev-platform/blob/main/web/frontend/assets/app/developer-v3/components/SidebarRight/common/LifecycleProgress/LifecycleProgress.tsx#L55)

@eliangcs

Reply:
ah ok, fair point! then yeah, this is maybe fine. steps are confusing.